### DISCUSSION TITLE
Why is microprofile-lra coupled to JAX-RS?

### DISCUSSION THREAD

**Discussion by jwgmeligmeyling** [eclipse/microprofile-lra#361] [Q&A]
The `@LRA` annotation specifically mentions:

>  The annotation *SHOULD* be applied to JAX-RS annotated methods otherwise it *MAY* have no effect.

Its exception handling is also bound to `Response.Status.Family` instead of `Exception` types.

What is the rationale behind this? Why can't it be possible for MessageDriven beans or EJB timers to participate in Long Running Actions? What would be the limitation of a more generic LRA API? Or perhaps an LRA SPI that allows different underlying implementations? I imagine i.e. providing the LRA ID through a JAX-RS Header Parameter is only syntactical, and it would just as well be possible to have some annotated parameter on an intercepted method. So what is the technical reason why limiting LRA to JAX-RS just made sense? What am I missing?

**Comment by mmusgrov** [comment]
MicroProfile LRA came from an implementation that we had of an earlier OASIS specification [1]. The MicroProfile core platform was built on a foundation of CDI, JAX-RS and JSON-P and the LRA spec seemed like a good fit for it and we wanted to only bring in those aspects that we had an implementation of (especially so because MicroProfile is an implementation first approach). The idea was to only bring proven ideas to the group. Of course now that MP users have had some experience of using LRA we can move on to broadening the scope of the specification.

[1] https://www.oasis-open.org/committees/document.php?document_id=12794
